### ISSUE TITLE
[Feature request] Automatic language detection

### ISSUE THREAD

**Issue by Hill-98** [dialect-app/dialect#213]
I have been using a translation software under Windows: [QTranslate](https://quest-app.appspot.com/), which is very good and has some great features.

One of my favorite features is: automatic language detection

QTranslate can set the first language and the second language, for example, the first language is set to Chinese, and the second language is set to English. When I type in Chinese, it will be translated into English, and when I type in other languages, it will be translated into Chinese.

This is a great feature, and I hope Dialect has it too.

**Comment by jeblad** [comment]
There is already an automatic detection of source language, but what you describe seems more like a feature to detect the direction of the translation? When I have set the source to Chinese and destination to English, but write English in the source field, then you want the translation direction to switch so destination field become Chinese?

**Comment by Hill-98** [comment]
Yes, for example, when I enter any non-Chinese language, it is automatically translated into Chinese. And when I type in Chinese, it is automatically translated into English.
